apps/dalmatiner_db/src/dalmatiner_tcp.erl

```diff
@@ -76,7 +75,20 @@ loop(Socket, Transport, State) ->
                                         dict = bkt_dict:new(Bucket,
                                                             State#state.n,
                                                             State#state.w)},
+                                {incomplete, <<>>});
+                {stream, Bucket, Delay, Resolution} ->
+                    lager:info("[tcp] Entering stream mode for bucket '~s' "
+                               "with max delay: ~p and resolution: ~pms",
+                               [Bucket, Delay, Resolution]),
+                    ok = dalmatiner_opt:set_resolution(Bucket, Resolution),
```

Comment:
I agree, but cannot recall a scenario where we send an error to the client over TCP.  For instance, if they send badly encoded data, resulting in a `badmatch` in dproto, we do not send any error information back to them.  
Printing the error is possible, but this feels like a client issue and would not help them much. 
Creating a new message type, specifically for errors, seems to be the only way to accomplish this.  I took a look through ranch docs, and it does not seem to use any underlying TCP semantics to signal failures.